scripts/issue/799/upgrade_vaults_v1_5.py

```diff
@@ -0,0 +1,175 @@
+HODLERS = [
+    "0xD14f076044414C255D2E82cceB1CB00fb1bBA64c",
+    "0xfe51263bd0d075dc5441e89ecd1f6d63ff41e02e",
+    "0xec4fcd1aca723f8456999c5f5d7479dbe9528c11",
+    "0x3cd3dba355776ac76a0485e728ead54092085da2",
+    "0xc3b1f7ab9fabd729cdf9e90ea54ec447f9464269",
+    "0x794783dcfcac8c1944727057a3208d8f8bb91506"
+]
+
+
+def main(queue="true", simulation="false"):
+    safe = GreatApeSafe(registry.eth.badger_wallets.dev_multisig)
+    safe.init_badger()
+
+    for key, address in VAULTS.items():
+        address = web3.toChecksumAddress(address)
+        if key in KEYS:
+            if queue == "true":
+                C.print(f"[green]Queuing upgrade for {key}[/green]")
+                safe.badger.queue_timelock(
+                    target_addr=DEV_PROXY,
+                    signature="upgrade(address,address)",
+                    data=encode_abi(
+                        ["address", "address"],
+                        [address, NEW_LOGIC],
+                    ),
+                    dump_dir="data/badger/timelock/upgrade_vaults_v1_5_fee_events/",
+                    delay_in_days=4,
+                )
+            else:
+                execute_timelock(
+                    safe.badger.timelock,
+                    "data/badger/timelock/upgrade_vaults_v1_5_fee_events/",
+                    key,
+                    address,
+                    simulation,
+                    safe
+                )
+        
+    safe.post_safe_tx()
+
+
+def execute_timelock(timelock, queueTx_dir, key, address, simulation, safe):
+    vault = interface.ITheVault(address)
+    if simulation == "false":
+        path = os.path.dirname(queueTx_dir)
+        directory = os.fsencode(path)
+
+        C.print(f"[green]Executing upgrade for {key}[/green]")
+
+        for file in os.listdir(directory):
+            filename = os.fsdecode(file)
+            if not filename.endswith(".json"):
+                continue
+            txHash = filename.replace(".json", "")
+
+            with open(f"{queueTx_dir}{filename}") as f:
+                tx = json.load(f)
+
+            # Get vault address from upgrade Tx
+            target_vault = web3.toChecksumAddress("0x" + tx["data"][24:64])
+
+            if vault.address == target_vault:
+                if timelock.queuedTransactions(txHash) == True:
+
+                    # Grab all variables before upgrade
+                    prev_available = vault.available()
+                    prev_gov = vault.governance()
+                    prev_keeper = vault.keeper()
+                    prev_guardian = vault.guardian()
+                    prev_strategist = vault.strategist()
+                    prev_treasury = vault.treasury()
+                    prev_token = vault.token()
+                    prev_symbol = vault.symbol()
+                    prev_balance = vault.balance()
+                    prev_totalSupply = vault.totalSupply()
+                    prev_getPricePerFullShare = vault.getPricePerFullShare()
```

Comment:
what is the reason writing all this out manually? instead of the programmatic approach such as https://github.com/Badger-Finance/badger-multisig/blob/09d414664dae385de3890e09015dfb3afe27cda7/scripts/issue/759/upgrade_uni_lp_geyser.py#L55-L61